src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -43,7 +43,7 @@ public void start() {
   */
   public User actAs(X x, User entity) throws AuthenticationException {
     User agent = (User) x.get("user");
-    User user = (User) userDAO_.find(entity.getId());
+    User user = (User) userDAO_.find_(x, entity.getId());
```

Comment:
Shouldn't this be the following?

```suggestion
    User user = (User) userDAO_.inX(x).find(entity.getId());
```

Reply:
This would be valid but since I'm passing an id into the method it doesn't make much of a difference. However I'll get into the habit of using inX since it avoids potential context related issues.